Bakker/models/bakker_koeken.py

```diff
@@ -0,0 +1,233 @@
+        seizoen_tag = self.env['bakker_koeken_tags'].search([('name', '=', 'Seizoen')], limit=1)
+        for record in self:
+            if seizoen_tag:
+                record.tags_ids = [(4, seizoen_tag.id)]
+            record.prijs_koek = record.prijs_koek * 1.15  # 15% prijsverhoging
+        return True
+    
+    def action_kwaliteitscontrole(self):
+        """Doe kwaliteitscontrole"""
+        import random
+        for record in self:
+            if random.choice([True, False]):  # 50% kans
+                
+                vers_tag = self.env['bakker_koeken_tags'].search([('name', '=', 'Vers')], limit=1)
+                if vers_tag:
+                    record.tags_ids = [(4, vers_tag.id)]
+        return {
+            'type': 'ir.actions.client',
+            'tag': 'display_notification',
+            'params': {
+                'title': '✅ Kwaliteitscontrole Voltooid',
+                'message': 'Kwaliteit gecontroleerd en goedgekeurd!',
+                'type': 'info',
+            }
+        }
+    
+    def action_view_low_stock(self):
+        """Toon koeken met lage voorraad"""
+        action = self.env.ref('Bakker.bakker_koeken_action').read()[0]
+        action['domain'] = [('voorraad_koek', '<=', 10)]
+        action['context'] = {'search_default_filter_voorraad_laag': 1}
+        return action
+    
+    def action_verkoop_rapport(self):
+        """Genereer verkoop rapport"""
+        return {
+            'name': 'Verkoop Rapport',
+            'type': 'ir.actions.act_window',
+            'res_model': 'bakker_koeken',
+            'view_mode': 'pivot,graph',
+            'context': {
+                'group_by': ['categorie_koek_id'],
+            }
+        }
+    
+    def action_verkoop_koek(self):
+        """Open wizard om koeken te verkopen"""
+        return {
+            'name': 'Koek Verkopen',
+            'type': 'ir.actions.act_window',
+            'res_model': 'bakker.verkoop.wizard',
+            'view_mode': 'form',
+            'target': 'new',
+            'context': {
+                'default_koek_id': self.id,
+                'default_prijs_per_stuk': self.prijs_koek,
+            }
+        }
+    
+    def action_view_verkopen(self):
+        """Toon alle verkopen voor deze koek"""
+        return {
+            'name': f'Verkopen voor {self.name_koek}',
+            'type': 'ir.actions.act_window',
+            'res_model': 'bakker_verkoop',
+            'view_mode': 'list,form',
+            'domain': [('koek_id', '=', self.id)],
+            'context': {'default_koek_id': self.id}
+        }
+    
+    def action_snelle_verkoop(self):
+        """Snelle verkoop - direct 1 koek verkopen"""
+        if self.voorraad_koek <= 0:
+            raise ValidationError("Geen voorraad beschikbaar!")
+        
+        # Zoek of maak een standaard walk-in klant
+        walk_in_klant = self.env['res.partner'].search([('name', '=', 'Walk-in klant')], limit=1)
```

Comment:
using a name for a search is never a good idea, it could change, another way would be to add a checkbox on the res.partner model and then search for the record that has this checkbox set

```suggestion
        walk_in_klant = self.env['res.partner'].search([('is_walk_in_client', '=', True)], limit=1)
```